ZenPacks/zenoss/Microsoft/Windows/utils.py

```diff
@@ -176,32 +185,49 @@ def getSQLAssembly():
         ASSEMBLY_2012,
         ASSEMBLY)
 
+    MSSQL_SMO = {9: MSSQL2005_SMO,
+                 10: MSSQL2008_SMO,
+                 11: MSSQL2012_SMO,
+                 12: MSSQL2012_SMO}
+
     ASSEMBLY_LOAD_ERROR = "write-host 'assembly load error'"
 
     sqlConnection = []
-    sqlConnection.append("try{")
-    sqlConnection.append(MSSQL2012_CONNECTION_INFO)
-    sqlConnection.append("}catch{")
-    sqlConnection.append("try{")
-    sqlConnection.append(MSSQL2008_CONNECTION_INFO)
-    sqlConnection.append("}catch{")
-    sqlConnection.append("try{")
-    sqlConnection.append(MSSQL2005_CONNECTION_INFO)
-    sqlConnection.append("}catch{")
-    sqlConnection.append(ASSEMBLY_LOAD_ERROR)
-    sqlConnection.append("}}}")
-
-    sqlConnection.append("try{")
-    sqlConnection.append(MSSQL2012_SMO)
-    sqlConnection.append("}catch{")
-    sqlConnection.append("try{")
-    sqlConnection.append(MSSQL2008_SMO)
-    sqlConnection.append("}catch{")
-    sqlConnection.append("try{")
-    sqlConnection.append(MSSQL2005_SMO)
-    sqlConnection.append("}catch{")
-    sqlConnection.append(ASSEMBLY_LOAD_ERROR)
-    sqlConnection.append("}}}")
+    if version not in [9, 10, 11, 12]:
+        sqlConnection.append("try{")
+        sqlConnection.append(MSSQL2012_CONNECTION_INFO)
+        sqlConnection.append("}catch{")
+        sqlConnection.append("try{")
+        sqlConnection.append(MSSQL2008_CONNECTION_INFO)
+        sqlConnection.append("}catch{")
+        sqlConnection.append("try{")
+        sqlConnection.append(MSSQL2005_CONNECTION_INFO)
+        sqlConnection.append("}catch{")
+        sqlConnection.append(ASSEMBLY_LOAD_ERROR)
+        sqlConnection.append("}}}")
+
+        sqlConnection.append("try{")
+        sqlConnection.append(MSSQL2012_SMO)
+        sqlConnection.append("}catch{")
+        sqlConnection.append("try{")
+        sqlConnection.append(MSSQL2008_SMO)
+        sqlConnection.append("}catch{")
+        sqlConnection.append("try{")
+        sqlConnection.append(MSSQL2005_SMO)
+        sqlConnection.append("}catch{")
+        sqlConnection.append(ASSEMBLY_LOAD_ERROR)
+        sqlConnection.append("}}}")
+    else:
+        sqlConnection.append("try{")
+        sqlConnection.append(MSSQL_CONNECTION_INFO.get(version))
+        sqlConnection.append("}catch{")
+        sqlConnection.append(ASSEMBLY_LOAD_ERROR)
+        sqlConnection.append("}")
+        sqlConnection.append("try{")
+        sqlConnection.append(MSSQL_SMO.get(version))
+        sqlConnection.append("}catch{")
+        sqlConnection.append(ASSEMBLY_LOAD_ERROR)
+        sqlConnection.append("}")
```

Comment:
Can these be string formatted so that the code structure is more apparent?
Otherwise comments would be helpful in understanding this.